tests/unit/test_emit_metrics.py

```diff
@@ -2,7 +2,60 @@
 from unittest import TestCase
 from unittest.mock import Mock
 
-from buildpack.runtime_components.metrics import FreeAppsMetricsEmitterThread
+from buildpack.runtime_components.metrics import (
+    FreeAppsMetricsEmitterThread,
+    PaidAppsMetricsEmitterThread,
+)
+
+
+class TestNegativeMemoryMetricsThrowError(TestCase):
+    def test_validating_bad_metrics(self):
+        m2ee_stats = {"memory": {"javaheap": -12345}}
+        with self.assertRaises(RuntimeError):
+            PaidAppsMetricsEmitterThread._sanity_check_m2ee_stats(m2ee_stats)
```

Comment:
I think that we are technically missing one test case here, the happy path where `m2ee_stats` only contains a positive int, and nothing else, right?


(The `test_non_ints_dont_cause_problems` function kinda does the same thing but not really; that is more checking that if you have a dict in `m2ee` tools as well as int values, no errors are raised)

Reply:
Good point. I have added a test for no memory pools, but good metrics.